core/src/test/java/com/linecorp/armeria/client/retry/RedirectRuleTest.java

```diff
@@ -0,0 +1,161 @@
+                return HttpResponse.of(headers);
+            });
+            sb.service("/loop2", (ctx, req) -> {
+                final ResponseHeaders headers = ResponseHeaders.of(HttpStatus.TEMPORARY_REDIRECT,
+                                                                   HttpHeaderNames.LOCATION, "/loop");
+                return HttpResponse.of(headers);
+            });
+        }
+    };
+
+    @BeforeEach
+    void setUp() {
+        counter.set(0);
+    }
+
+    @Test
+    void redirect_successWithRetry() {
+        final AggregatedHttpResponse res = sendRequest(2);
+        assertThat(res.contentUtf8()).contains("fooRedirect");
+    }
+
+    private static AggregatedHttpResponse sendRequest(int totalAttempts) {
+        final RetryConfig<HttpResponse> config =
+                RetryConfig.builder(RetryRule.redirect().orElse(RetryRule.onServerErrorStatus()))
+                           .maxTotalAttempts(totalAttempts)
+                           .build();
+        final WebClient client = WebClient.builder(server.httpUri())
+                                          .decorator(RetryingClient.newDecorator(config))
+                                          .build();
+        return client.get("/foo").aggregate().join();
+    }
+
+    @Test
+    void redirect_failExceedingTotalAttempts() {
+        final AggregatedHttpResponse res = sendRequest(1);
+        assertThat(res.status()).isSameAs(HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+
+    @Test
+    void httpsRedirect() {
+        final WebClient client = WebClient.builder(server.httpUri())
+                                          .factory(ClientFactory.insecure())
+                                          .decorator(RetryingClient.newDecorator(RetryRule.redirect()))
+                                          .build();
+        final AggregatedHttpResponse join = client.get("/https").aggregate().join();
+        assertThat(join.contentUtf8()).contains("httpsRedirection");
+    }
+
+    @Test
+    void seeOtherHttpMethodChangedToGet() {
+        final WebClient client = WebClient.builder(server.httpUri())
+                                          .decorator(RetryingClient.newDecorator(RetryRule.redirect()))
+                                          .build();
+        final AggregatedHttpResponse join = client.post("/seeOther", "hello!").aggregate().join();
+        assertThat(join.contentUtf8()).contains("seeOtherRedirection");
+    }
+
+    @Test
+    void redirectLoopsException() {
+        final WebClient client = WebClient.builder(server.httpUri())
+                                          .decorator(RetryingClient.newDecorator(RetryRule.redirect()))
+                                          .build();
+        assertThatThrownBy(() -> client.get("/loop").aggregate().join())
+                .hasMessageContainingAll("originalPath: /loop, redirect paths:", "loop1", "loop2");
```

Comment:
Do we need to normalize paths? Wouldn't it better if the redirect paths were `/loop1` and `/loop2`?

Reply:
It's actually `/loop1` and `/loop2`. Let me add `/` to each path. 😄